exporter/googlecloudexporter/config.go

```diff
@@ -0,0 +1,175 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package googlecloudexporter
+
+import (
+	"os"
+
+	"github.com/GoogleCloudPlatform/opentelemetry-operations-collector/processor/normalizesumsprocessor"
+	"github.com/mitchellh/mapstructure"
+	"github.com/observiq/observiq-otel-collector/processor/resourceattributetransposerprocessor"
+	gcp "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/googlecloudexporter"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourceprocessor"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/processor/batchprocessor"
+	"go.uber.org/multierr"
+)
+
+const (
+	defaultMetricPrefix = "workloads.googleapis.com"
+	defaultUserAgent    = "observIQ-otel-collector"
+	defaultLocation     = "us-east1-b"
+	genericNodeResource = "generic_node"
+)
+
+// Config is the config the google cloud exporter
+type Config struct {
+	config.ExporterSettings `mapstructure:",squash"`
+	Location                string                                       `mapstructure:"location"`
+	Namespace               string                                       `mapstructure:"namespace"`
+	GCPConfig               *gcp.Config                                  `mapstructure:",squash"`
+	BatchConfig             *batchprocessor.Config                       `mapstructure:"batch"`
+	NormalizeConfig         *normalizesumsprocessor.Config               `mapstructure:"normalize"`
+	DetectorConfig          *resourcedetectionprocessor.Config           `mapstructure:"detector"`
+	AttributerConfig        *resourceprocessor.Config                    `mapstructure:"attributer"`
+	TransposerConfig        *resourceattributetransposerprocessor.Config `mapstructure:"transposer"`
+}
+
+// Validate validates the config
+func (c *Config) Validate() error {
+	var err error
+	err = multierr.Append(err, c.GCPConfig.Validate())
+	err = multierr.Append(err, c.BatchConfig.Validate())
+	err = multierr.Append(err, c.NormalizeConfig.Validate())
+	err = multierr.Append(err, c.DetectorConfig.Validate())
+	err = multierr.Append(err, c.AttributerConfig.Validate())
+	err = multierr.Append(err, c.TransposerConfig.Validate())
+	return err
+}
+
+// createDefaultConfig creates the default config for the exporter
+func createDefaultConfig() config.Exporter {
+	defaultNamespace, _ := os.Hostname()
```

Comment:
Is this what namespace should really be by default? I'm not really sure, semantically, what the namespace should be, but the resource detection processor + node_id already fills this role of labelling w/ the hostname, right?

Reply:
Im leaning more toward the source name ("postgresql", "mysql", etc) but I believe it is not currently possible unless we set a resource somewhere. I think hostname is okay.

Google uses `namespace` to organize data on the backend. They instructed us to use "something" other than default.

Reply:
If something other than `default` is needed, then this is probably OK then. It just seems redundant from a user perspective, wasn't aware of the technical reason.